pint_server/app.py

```diff
@@ -425,11 +439,39 @@ def get_provider_servers(provider):
             for server in servers]
 
 
+def trim_images_payload(images):
+    payload_size = jsonify(images=images).content_length
+
+    if payload_size >  MAX_PAYLOAD_SIZE:
+        # NOTE: assuming the size of the entries are evenly distributed, we
+        # determine the percentage of entries to trim from the end of the list,
+        # as the list is sorted in decending order by publishedon date, by
+        # calculating the percentage over the maximum payload size. Then
+        # trim the same percentage off the list, rounding up to be safe.
+        trim_size  = math.ceil(
+            ((payload_size - MAX_PAYLOAD_SIZE) / payload_size) * len(images))
+        images = images[:-trim_size]
+
+        # Now make sure we don't have partial data by finished triming all the
+        # images from all regions that have the same publishedon date that of
+        # the last image that got trimmed. But if the last entry is already at
+        # the borard, this algorithm will trim off the entire data set with
+        # the same publishedon date, which is not a concern for us.
+        publishedon = images[-1]['publishedon']
```

Comment:
The problem with this is that if the trim happens to land on the boundary then we trim images unnecessarily. The `publishedon` date that drives the trim really has to come from the first element of the slice that is getting trimmed off, i.e. 
```
last_publishedon = images[trim_size-1]['publishedon']
```
and this needs to be before

```
images = images[:-trim_size]
```

and then the while loop changes to

```
while images[-trim_size]['publishedon'] == last_publishedon
```
This avoids trimming a whole set of images if the trim happens to land on a "publishedon" boundary.

Reply:
I've added the boundary scenario in the code comment, that trimming extra is not a concern. But I can update the code to account for the boundary scenario.